core/shared/src/main/scala/io/circe/Encoder.scala

```diff
@@ -22,7 +21,9 @@ trait Encoder[A] extends Serializable {
   /**
    * Creates a new instance by applying a function to a value of type `B` before encoding as an `A`.
    */
-  final def contramap[B](f: B => A): Encoder[B] = Encoder.instance(b => apply(f(b)))
+  final def contramap[B](f: B => A): Encoder[B] = new Encoder[B] {
```

Comment:
@vkostyukov Not primarily—the issue was that using `instance` in the encoders for primitives was causing extra boxing / unboxing, and I decided that a blanket no-`instance`-in-core rule would be most consistent. For the generic methods it's nice to save those allocations as well, I guess.